app/models/project.rb

```diff
@@ -10,17 +10,21 @@ class Project < ApplicationRecord
   has_many_attached :images
   has_many_attached :videos
   has_many_attached :audio
+  has_one :school_project, dependent: :destroy
 
   accepts_nested_attributes_for :components
 
   before_validation :check_unique_not_null, on: :create
+  before_validation :create_school_project_if_needed
```

Comment:
I think it would be safer to use `after_save` for this, so we can ensure the record has passed validation and has actually been saved into the database

Reply:
Hmmm, I guess in that case we wouldn't be able to validate the school project in the same way as we are currently 🤔